plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts

```diff
@@ -0,0 +1,225 @@
+    return new InstrumentationNodeModuleFile<typeof knex>(
+      `knex/${basePath}/client.js`,
+      constants.SUPPORTED_VERSIONS,
+      (Client: any, moduleVersion) => {
+        api.diag.debug(
+          `Applying client.js patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'queryBuilder',
+          this.storeContext.bind(this)
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'schemaBuilder',
+          this.storeContext.bind(this)
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'raw',
+          this.storeContext.bind(this)
+        );
+        return Client;
+      },
+      (Client: any, moduleVersion) => {
+        api.diag.debug(
+          `Removing client.js patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this._unwrap(Client.prototype, 'queryBuilder');
+        this._unwrap(Client.prototype, 'schemaBuilder');
+        this._unwrap(Client.prototype, 'raw');
+        return Client;
+      }
+    );
+  }
+
+  private createQueryWrapper(moduleVersion?: string) {
+    const instrumentation = this;
+    return function wrapQuery(original: Function) {
+      return function wrapped_logging_method(this: any, query: any) {
+        const config = this.client.config;
+
+        const maxLen = (
+          instrumentation._config as types.KnexInstrumentationConfig
+        ).maxQueryLength!;
+        const attributes: api.SpanAttributes = {
+          'knex.version': moduleVersion,
+          [SemanticAttributes.DB_SYSTEM]: utils.mapSystem(config.client),
+        };
+        if (maxLen !== 0) {
+          attributes[SemanticAttributes.DB_STATEMENT] = utils.limitLength(
+            query?.sql,
+            maxLen
+          );
+        }
+        const table = this.builder?._single?.table;
+        if (table) {
+          attributes[SemanticAttributes.DB_SQL_TABLE] = table;
+        }
+        // `method` actually refers to the knex API method - Not exactly "operation"
+        // in the spec sense, but matches most of the time.
+        const operation = query?.method;
+        if (operation) {
+          attributes[SemanticAttributes.DB_OPERATION] = operation;
+        }
+        const user = config?.connection?.user;
+        if (user) {
+          attributes[SemanticAttributes.DB_USER] = config.connection.user;
+        }
+        const name =
+          config?.connection?.filename || config?.connection?.database;
+        if (name) {
+          attributes[SemanticAttributes.DB_NAME] = name;
+        }
+
+        const parent = this.builder[contextSymbol];
+        const span = instrumentation.tracer.startSpan(
+          utils.getName(name, operation, table),
+          {
+            attributes,
+          },
+          parent
+        );
+
+        return original
```

Comment:
I did not setup a real environment, so not sure about what I'm saying, but from the docs likes like `knex` is actually running the query itself with the underlying driver.
In this case, maybe we should use `context.with` here so the underlying driver span will be the child of `knex` span?

Reply:
Tested it on my setup and the `pg` driver span is indeed not the child of `knex`. IMO, we should wrap the call to `original.apply` with `context.with(...)`

Reply:
I'll def make that change and double-test that! Thanks!

Reply:
![knex-pg](https://user-images.githubusercontent.com/1334536/121012430-6221a900-c7a0-11eb-8849-8474bc67d075.png)
Looks better now!